### PR TITLE
Support for PCAP analysis + Suricata analysis + updated PEfile

### LINKED ISSUE #679: Support for PCAP and dedicated analyzers
We could add PCAPs as a new supported artifacts to analyze. This could be extremely useful for cyber security analysts that usually deal with this kind of artifacts

The new analyzers could be:

- zeek
- suricata
- packettotal
- etc

### PR DESCRIPTION
**Pull request by mlodic**
# Description

Added support for PCAP analyzers and first PCAP analyzer: Suricata

## Related issues
closes #679 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [ ] The pull request is for the branch `develop`
- [ ] A new analyzer or connector was added, in which case:
    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](./Advanced-Usage.md) was updated (in case the analyzer/connector provides additional optional configuration).
    - [ ] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
    - [ ] If the analyzer/connector requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] If a File analyzer was added, its name was explicitly defined in [test_file_scripts.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_file_scripts.py) (not required for Observable Analyzers).
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [ ] The tests gave 0 errors.
- [ ] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)
  
### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

# Real World Example

Please delete if the PR is for bug fixing.
Otherwise, please provide the resulting raw JSON of a finished analysis (and, if you like, a screenshot of the results). This is to allow the maintainers to understand how the analyzer works.

### CODE REVIEW

**Comment by mlodic** [comment]
image

File: api_app/analyzers_manager/file_analyzers/suricata.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        report["signatures"] = self.signatures
+        if not self.extended_logs:
+            del report["data"]
+
+        return report
+
+    def _add_item_to_signatures(self, item, key, signature_name):
+        item = str(item)
+        if item and item not in self.signatures[signature_name][key]:
+            self.signatures[signature_name][key].append(item)
```
Comment (0ssigeno) [inline api_app/analyzers_manager/file_analyzers/suricata.py:82]: supposing that self.signatures[signature_name][key] is a `set`, you could just use `add`.

File: integrations/pcap_analyzers/app.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                    res["report"]["data"].append(json.loads(line))
+                except json.JSONDecodeError as e:
+                    logger.debug(e)
+                    res["report"]["data"].append(fp.read())
+
+    # 3. set final result after modifications
+    future._result = res
+
+    # 4 remove the log file
+    shutil.rmtree(directory)
```
Comment (0ssigeno) [inline integrations/pcap_analyzers/app.py:69]: Should check that directory is not `None` or you would have a `TypeError`

File: integrations/pcap_analyzers/check_pcap.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            )
+            self.filename = str(args.filename)
+            self.job_id = args.job_id
+            self.md5 = args.file_md5
+            self.reload_rules = args.reload_rules
+            self.analysis_dir = f"/tmp/eve_{self.job_id}"
+            self.eve_file = self.analysis_dir + "/eve.json"
+            self.check_pcap()
+        except Exception as e:
+            logger.exception(e)
```
Comment (0ssigeno) [inline integrations/pcap_analyzers/check_pcap.py:55]: Do we need to try/catch all this block + generic exception? Can't we limit the scope?

File: integrations/pcap_analyzers/app.py (hunk 2 above)
Comment (mlodic) [reply integrations/pcap_analyzers/app.py:69]: that exists for sure cause this a callback from the check_pcap.py where that directory is created

File: api_app/analyzers_manager/file_analyzers/suricata.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            if signature_name not in self.signatures:
+                self.signatures[signature_name] = {
+                    "alerts_triggered": 0,
+                    "protocols": [],
+                    "src_ips": [],
+                    "dest_ips": [],
+                    "src_ports": [],
+                    "dest_ports": [],
+                }
+                self.signatures[signature_name].update(alert)
```
[the remark this answers is not in the document]
Comment (mlodic) [reply api_app/analyzers_manager/file_analyzers/suricata.py:52]: yeah they are different

File: api_app/analyzers_manager/dataclasses.py [hunk 5]
```diff
@@ -44,8 +44,16 @@ def is_observable_type_supported(self, observable_classification: str) -> bool:
...
     def is_filetype_supported(self, file_mimetype: str) -> bool:
+        # PCAPs are not classic files. They should not leverage the default behavior.
+        # We should execute them only if the analyzer specifically support them.
+        special_pcap_mimetype = "application/vnd.tcpdump.pcap"
+        if (
+            file_mimetype == special_pcap_mimetype
+            and special_pcap_mimetype not in self.supported_filetypes
+        ):
+            return False
+        # base case: empty lists means supports all
```
[the remark this answers is not in the document]
Comment (mlodic) [reply api_app/analyzers_manager/dataclasses.py:55]: PCAPs are a file exception and won't be any other like them. I don't think there is need to generalize this.

File: integrations/pcap_analyzers/check_pcap.py (hunk 3 above)
Comment (mlodic) [reply integrations/pcap_analyzers/check_pcap.py:55]: if you don't do that, unexpected exceptions won't be logged and you'll never know that something has broken

**Pull request merged**